core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -368,23 +383,22 @@ private void applyOutstandingWalTransactions(
                             isTerminating = runStatus.isTerminating();
                             final long added = processWalCommit(
                                     writer,
+                                    walId,
                                     tempPath,
                                     segmentTxn,
                                     operationCompiler,
                                     seqTxn,
+                                    commitTimestamp,
                                     isTerminating,
                                     transactionMeta,
                                     iTransaction * TXN_METADATA_LONGS_SIZE
                             );
 
                             if (added > -1L) {
-                                long timeDelta = microClock.getTicks() - start;
-                                insertTimespan += timeDelta;
+                                insertTimespan += microClock.getTicks() - start;
                                 rowsAdded += added;
                                 iTransaction++;
-                                long physicallyWrittenRowsSinceLastCommit = writer.getPhysicallyWrittenRowsSinceLastCommit();
-                                physicalRowsAdded += physicallyWrittenRowsSinceLastCommit;
-                                metrics.addApplyRowsWritten(added, physicallyWrittenRowsSinceLastCommit, timeDelta);
```

Comment:
Where this line has gone?
`metrics` gathers gauges for Prometheus, it is also needed.

Reply:
it is inside of `processWalCommit`, close to telemetry event